packages/dds/tree/src/util/utils.ts

```diff
@@ -54,6 +54,20 @@ export function fail(message: string): never {
 	throw new Error(message);
 }
 
+/**
+ *
+ * Use this function to assert that an array index is not out-of-bounds.
+ * @example
+ * ```ts
+ * // We know that `numberArray` has four elements in it, so this is safe.
+ * const n: number = numberArray[3] ?? oob();
+ * ```
+ * @internal
```

Comment:
No need to give this an access tag since it is not exported out of the package. If it were being exported out of the top-level index.ts in the tree package (packages/dds/tree/src/index.ts) then it would need a tag.